.github/workflows/post-merge-dynamo-to-builb.yml

```diff
@@ -0,0 +1,53 @@
+name: Deploy Main to Build Env L2Dynamo - cic-cri-ddb
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch
+  push:
+    branches:
+      - main
+    paths:
+      - 'infra-l2-dynamo/**'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+env: # Only adding the variables in that are required for 
+  AWS_REGION: eu-west-2
+
+jobs:
+  deploy-to-dev:
+    name: Validate & Deploy L2Dynamo template to Build Envrionment
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./infra-l2-dynamo/**
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+
+      - name: Setup SAM CLI
+        uses: aws-actions/setup-sam@v2
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          role-to-assume: ${{ secrets.BUILD_CRI_V1_GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: SAM Validate
+        run: sam validate --region ${{ env.AWS_REGION }} -t template.yaml
+
+      - name: SAM Build
+        run: sam build -t template.yaml
+
+      - name: Deploy SAM app
+        uses: alphagov/di-devplatform-upload-action@v3
+        with:
+            artifact-bucket-name: ${{ secrets.BUILD_CRI_V1_ARTIFACT_SOURCE_BUCKET_NAME }}
```

Comment:
Do you need to pass a signing-key-arn into dev? (This may be a problem with documentation, but https://github.com/alphagov/di-devplatform-upload-action it implies that `signing-profile-name` is required.)

I don't think it should be required, but let's check if the docs are up to date.

Reply:
I think Viveak had a chat with Dev and he mentioned not to code signing step. See our slack chat 
https://gds.slack.com/archives/C04PQE5D5V4/p1676908143827899?thread_ts=1676907031.398039&cid=C04PQE5D5V4